src/main/scala/com/cloudera/sparkts/models/KNNRegression.scala

```diff
@@ -0,0 +1,170 @@
+/**
+  * Copyright (C) 2017 Verizon. All Rights Reserved.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *     http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package com.cloudera.sparkts.models
+
+import org.apache.spark.mllib.KernelType._
+import org.apache.spark.mllib.Kernel
+import org.apache.spark.mllib.linalg.{Vector, Vectors}
+import scala.collection.mutable.ArrayBuffer
+import java.util.PriorityQueue
+import java.util.Comparator
+import scala.collection.JavaConverters._
+
+class BoundedPriorityQueue[E](k: Int,
+                              comp: Comparator[E]) extends PriorityQueue[E](k, comp) {
+  override def add(elem: E): Boolean = {
+    if (size() < k) {
+      offer(elem)
+      true
+    } else {
+      val head = peek()
+      if (head != null && comp.compare(elem, head) > 0) {
+        poll()
+        offer(elem)
+        return true
+      }
+      return false
+    }
+  }
+}
+
+/**
+  * @author debasish83, xiangzhe, santanu.das
+  */
+//TODO: history and query can be combined into a vector with offset giving queries
+//TODO: distance and point both should be covered by Neighbor
+case class Neighbor(history: Vector, query: Double)
+
+class NeighborOrder extends Ordering[(Int, Double)] {
+  override def compare(x: (Int, Double), y: (Int, Double)): Int = {
+    x._2 compare y._2
+  }
+}
+
+class KNNRegressionModel extends TimeSeriesModel {
+  override def addTimeDependentEffects(ts: Vector, dest: Vector) = ???
+
+  override def removeTimeDependentEffects(ts: Vector, dest: Vector) = ???
+}
+
+object KNNRegression {
+  def maxnorm(timeseries: Array[Double]): Double = {
+    val max = timeseries.max
+    var i = 0
+    while (i < timeseries.length) {
+      timeseries(i) /= max
+      i += 1
+    }
+    return max
+  }
+
+  def nearestNeighbors(timeseries: Array[Double],
+                       featureDim: Int,
```

Comment:
For multiline indentation, we use the convention used here: https://github.com/sryza/spark-timeseries/blob/master/src/main/scala/com/cloudera/sparkts/models/ARIMA.scala#L80